src/pk.c

```diff
@@ -259,15 +274,26 @@ static int wolfssl_print_number(WOLFSSL_BIO* bio, mp_int* num, const char* name,
     if (ret == 1) {
         /* Start first digit line with spaces.
          * Writing out zeros ensures number is a positive value. */
-        li = XSNPRINTF(line, sizeof(line) - 1, PRINT_NUM_INDENT "%s",
+        li = XSNPRINTF(line, sizeof(line), PRINT_NUM_INDENT "%s",
             mp_leading_bit(num) ?  "00:" : "");
+        if (li >= (int)sizeof(line)) {
+            WOLFSSL_MSG("Buffer overflow formatting spaces");
+            ret = 0;
+        }
     }
 
     /* Put out each line of numbers. */
     for (i = 0; (ret == 1) && (i < rawLen); i++) {
-        /* Check for a complete line. */
-        if (li >= PRINT_NUM_MAX_DIGIT_LINE) {
-            /* More bytes coming so append carriage return. */
+        /* Encode another byte as 2 hex digits and append colon. */
+        int len_wanted = XSNPRINTF(line + li, sizeof(line) - li, "%02x:",
```

Comment:
How does len_wanted get to be something other than 3?
What happens when the last byte is the last on the line? Are we going to put out the ':' anyway?

Reply:
>How does len_wanted get to be something other than 3?

It couldn't but there's no need to break the safe code pattern.

>What happens when the last byte is the last on the line? Are we going to put out the ':' anyway?

That's what the incumbent code was trying to do, now that you mention it, but if you want me to neaten things up further I can do that tomorrow.  I think it's probably good enough as-is.

Reply:
@SparkiDev I did some more code review and ran some experiments -- the latest commit actually DTRT, w00t!  if the octets divide evenly by the 15 octets/line rendering format, then the main loop will quit without rendering the final line, and the code at lines 313-319 will put it out, sans spurious ":".